src/testRunner/unittests/services/utilities.ts

```diff
@@ -0,0 +1,20 @@
+import * as ts from "../../_namespaces/ts";
+
+describe("unittests:: services:: utilities", () => {
+    describe("Test findPrecedingMatchingToken,", () => {
+        it("should not infinite loop finding opening brace", () => {
+            const sourceFile = ts.createSourceFile("file.ts", `/// <reference path="./compiler.d.ts" />
+
+(/** @window => {
+  /** @type {Abcd123} */
+  const core = window.Abcd.core;
+})();`, ts.ScriptTarget.ESNext, /*setParentNodes*/ true);
+            // can't use ts.getTokenAtPosition because it returns back the wrong token
+            const param = ts.forEachChildRecursively(sourceFile, node => node.kind === ts.SyntaxKind.Parameter ? node : undefined)!;

```

Comment:
what parameter is there in the example code?

Reply:
oh, `window`??

Reply:
No idea, I just copied and pasted the test from the issue...